src/utils/misc-utils.ts

```diff
@@ -203,7 +203,11 @@ export const getDereffedSchema = async (val: any) => {
 };
 
 export const configCanBeEmpty = (schema: any) => {
-    return Boolean(!schema?.properties || isEmpty(schema?.properties));
+    if (!schema) {
+        return false;
+    }
+
+    return Boolean(!schema.properties || isEmpty(schema.properties));
```

Comment:
Pretty sure this introduced the ability for a race condition

https://github.com/estuary/ui/pull/1230/files#r1729428622
![image](https://github.com/user-attachments/assets/b35fae6c-e87f-407a-82d8-23d970169c3b)